### PR TITLE
core/state, core/types: improve method receiver names

### PR DESCRIPTION
**Pull request by qshuai**

### CODE REVIEW

**Comment by qshuai** [comment]
@karalabe @holiman PTAL

File: core/state/statedb.go [hunk 1]
```diff
@@ -492,87 +492,87 @@ func (db *StateDB) ForEachStorage(addr common.Address, cb func(key, value common
...
 
 // GetRefund returns the current value of the refund counter.
-func (self *StateDB) GetRefund() uint64 {
-	return self.refund
+func (db *StateDB) GetRefund() uint64 {
+	return db.refund
 }
 
-// Finalise finalises the state by removing the self destructed objects
+// Finalise finalises the state by removing the db destructed objects
```
Comment (karalabe) [inline core/state/statedb.go]: Please revert the comment (i.e. self destructed).

Comment (karalabe) [reply core/state/statedb.go]: Otherwise LGTM

Comment (qshuai) [reply core/state/statedb.go]: @karalabe Thanks, have been fixed.

Comment (qshuai) [reply core/state/statedb.go]: @karalabe Please review the revised commit!

**Comment by qshuai** [comment]
> In general looks fine, but regarding `StateDB` I don't like changing `self` to `db`, since `StateDB` has an internal `db`. That makes it less readable, when there are constructs like `db.db` and generally at-first-glance not obvious if something is operating against the internal `db` or against the statedb itself. If you revert that change, I'm happy with it.

Yes, `db.db` likes puzzled.

**Comment by fjl** [comment]
We will re-apply this change after the 1.9.0 release.

**Pull request closed without merging**